homeassistant/components/tahoma/binary_sensor.py

```diff
@@ -0,0 +1,97 @@
+"""Support for Somfy TaHoma binary sensors."""
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_OCCUPANCY,
+    DEVICE_CLASS_OPENING,
+    DEVICE_CLASS_SMOKE,
+    DOMAIN as BINARY_SENSOR,
+    BinarySensorEntity,
+)
+
+from .const import DOMAIN
+from .tahoma_entity import TahomaEntity
+
+CORE_ASSEMBLY_STATE = "core:AssemblyState"
+CORE_BUTTON_STATE = "core:ButtonState"
+CORE_CONTACT_STATE = "core:ContactState"
+CORE_GAS_DETECTION_STATE = "core:GasDetectionState"
+CORE_OCCUPANCY_STATE = "core:OccupancyState"
+CORE_OPENING_STATE = "core:OpeningState"
+CORE_OPEN_CLOSED_TILT_STATE = "core:OpenClosedTiltState"
+CORE_RAIN_STATE = "core:RainState"
+CORE_SMOKE_STATE = "core:SmokeState"
+CORE_THREE_WAY_HANDLE_DIRECTION_STATE = "core:ThreeWayHandleDirectionState"
+CORE_VIBRATION_STATE = "core:VibrationState"
+CORE_WATER_DETECTION_STATE = "core:WaterDetectionState"
+
+DEVICE_CLASS_BUTTON = "button"
+DEVICE_CLASS_GAS = "gas"
+DEVICE_CLASS_RAIN = "rain"
+DEVICE_CLASS_WATER = "water"
+
+IO_VIBRATION_DETECTED_STATE = "io:VibrationDetectedState"
+
+STATE_OPEN = "open"
+STATE_PERSON_INSIDE = "personInside"
+STATE_DETECTED = "detected"
+STATE_PRESSED = "pressed"
+
+TAHOMA_BINARY_SENSOR_DEVICE_CLASSES = {
+    "AirFlowSensor": DEVICE_CLASS_GAS,
+    "CarButtonSensor": DEVICE_CLASS_BUTTON,
+    "ContactSensor": DEVICE_CLASS_OPENING,
+    "MotionSensor": DEVICE_CLASS_MOTION,
+    "OccupancySensor": DEVICE_CLASS_OCCUPANCY,
+    "RainSensor": DEVICE_CLASS_RAIN,
+    "SirenStatus": DEVICE_CLASS_OPENING,
+    "SmokeSensor": DEVICE_CLASS_SMOKE,
+    "WaterDetectionSensor": DEVICE_CLASS_WATER,
+    "WaterSensor": DEVICE_CLASS_WATER,
+    "WindowHandle": DEVICE_CLASS_OPENING,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the TaHoma sensors from a config entry."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    coordinator = data["coordinator"]
+
+    entities = [
+        TahomaBinarySensor(device.deviceurl, coordinator)
+        for device in data["platforms"].get(BINARY_SENSOR)
```

Comment:
```suggestion
        for device in data["platforms"][BINARY_SENSOR]
```

Reply:
`data["platforms"][BINARY_SENSOR]` is optional, since not all Somfy TaHoma devices will require a binary_sensor. Would it be okay to keep the `get(BINARY_SENSOR)` in this scenario?

Reply:
We only forward the entry to the platform if the platform is in `platforms`. Did I miss something?

Reply:
No, you are right. Oops! :D

Reply:
Fixed.